### ISSUE TITLE
bytes and packets counters are zero for netflow version 9 from cisco asa

### ISSUE THREAD

**Issue by snatch2013** [pmacct/pmacct#182]
Hello,
Could you please suggest how to resolve the problem with NetFlow v9 from the Cisco ASA?
Asa ios version is 9.1(7)19, nfacctd 1.7.1-git (20180125-01), before used 1.5.2. 

it is mentioned in the release notes that they made some changes after ios 8.4.5:

Resolution:
Nothing from our side, customer needs to contact the collector support and get a version which supports the ASA side changes.

The same has been acknowledged and fixed by multiple vendors:

Solarwinds: http://thwack.solarwinds.com/thread/52901 (HotFix 3 for NTA 3.10)

Plixer: http://www.plixer.com/blog/ipfix-2/cisco-asa-8-45-netflow-support/ (Scrutinizer version 10.1)

ManageEngine: http://blogs.manageengine.com/2012/12/03/cisco-asa-8-45-and-above-netflow-updates/ (Patch for NetFlow Analyzer 9.7 Build 9700)

**Comment by paololucente** [comment]
Hi @snatch2013 , thanks for opening this issue. Can you please send here or via unicast email (preferred) a sample of the NSEL traffic in libpcap format, (ie. captured with tcpdump)? This will help to visualise the actual changes to the flow format and handle them. Thanks Paolo

**Comment by snatch2013** [comment]
Hello Paolo,

I have sent an email to you.

**Comment by paololucente** [comment]
Dear @snatch2013 : thanks for following up privately. Setting 'tmp_asa_bi_flow: true' allows to gather bi-directional flow counters for Cisco ASA NSEL. Please confirm it working for you, caveats described here: https://github.com/pmacct/pmacct/blob/master/CONFIG-KEYS#L2576-L2582

**Comment by snatch2013** [comment]
Hello Paolo,

sorry for the late answer. I have just tried to add tmp_asa_bi_flow: true
to the nfacct.conf file and restarted the nfacctd service, but nothing
changed, still zeroes.
   Maybe I did something wrong? Shouldn't I use this command specifically
for the plugin - tmp_asa_bi_flow[asa]: true?

Regards,
Vladimir Stepanov




2018-01-29 17:34 GMT+01:00 Paolo Lucente <notifications@github.com>:
